src/components/common/productCard/productCard.tsx

```diff
@@ -0,0 +1,16 @@
+import {CardProps} from "../../../types";
+import {useParams} from "react-router-dom";
+import api from "../../../api/api.json"
+import {Card} from "../card";
+import css from "./productcCard.module.css"
+
+export const ProductCard = () => {
+
+    const {product} = useParams()
+    const sort = api.find((item: CardProps)=> item.label === product)
+    return (
+        <div className={css.product_card}>
+            <Card  label={sort!.label} img={sort!.img} description={sort!.description} price={sort!.price} />
```

Comment:
Тут и вообще везде в компонентах надо убрать восклицательные знаки. Пока бек не вернут товар или категорию, у тебя на клиенте его не будет, а значит надо показывать лоадер
```suggestion
            <Card  label={sort.label} img={sort.img} description={sort.description} price={sort.price} />
```